azure/packages/test/scenario-runner/src/logger.ts

```diff
@@ -96,6 +111,11 @@ class ScenarioRunnerLogger extends TelemetryLogger implements ITelemetryBuffered
 			}
 		}
 
+		// We want to emit any events that match ScenarioRunnerTelemetryEventNames
+		if (telemetryEventName !== undefined) {
+			this.events.emit(telemetryEventName, { ...event, originalEventName });
```

Comment:
The function itself is a 'transform' function. Maybe we rename it to something better, if we are emitting events.

Reply:
Curious on the purpose of emitting events here?

Reply:
Ok I see the reason.

Reply:
This is in the `send` function